Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/PrimerHeadlessUniversalCheckout.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  PrimerHeadlessUniversalCheckout.swift
+//  PrimerSDK
+//
+//  Created by Evangelos on 28/1/22.
+//
+
+#if canImport(UIKit)
+
+import UIKit
+
+public class PrimerHeadlessUniversalCheckout {
+    
+    public static let current = PrimerHeadlessUniversalCheckout()
+    
+    public weak var delegate: PrimerHeadlessUniversalCheckoutDelegate?
+    public weak var uiDelegate: PrimerHeadlessUniversalCheckoutUIDelegate?
+    private(set) public var clientToken: String?
+
+    private var apiConfigurationModule: PrimerAPIConfigurationModuleProtocol = PrimerAPIConfigurationModule()
+    private let unsupportedPaymentMethodTypes: [String] = [
+        PrimerPaymentMethodType.adyenBlik.rawValue,
+        PrimerPaymentMethodType.adyenDotPay.rawValue,
+        PrimerPaymentMethodType.adyenIDeal.rawValue,
+        PrimerPaymentMethodType.goCardless.rawValue,
+        PrimerPaymentMethodType.primerTestKlarna.rawValue,
+        PrimerPaymentMethodType.primerTestPayPal.rawValue,
+        PrimerPaymentMethodType.primerTestSofort.rawValue,
+        PrimerPaymentMethodType.xfersPayNow.rawValue,
+    ]
+    
+    fileprivate init() {}
+    
+    public func start(
+        withClientToken clientToken: String,
+        settings: PrimerSettings? = nil,
+        delegate: PrimerHeadlessUniversalCheckoutDelegate? = nil,
+        uiDelegate: PrimerHeadlessUniversalCheckoutUIDelegate? = nil,
+        completion: @escaping (_ paymentMethods: [PrimerHeadlessUniversalCheckout.PaymentMethod]?, _ err: Error?) -> Void
+    ) {
+        PrimerInternal.shared.intent = .checkout
+        
+        if delegate != nil {
+            PrimerHeadlessUniversalCheckout.current.delegate = delegate
+        }
+        
+        if PrimerHeadlessUniversalCheckout.current.delegate == nil {
+            print("WARNING!\nPrimerHeadlessUniversalCheckout delegate has not been set, and you won't be able to receive the Payment Method Token data to create a payment.")
```

Comment:
It's not just about the payment method token data: you won't be able to receive the list of payment methods either.
Tbh, I think we should just throw if `delegate` is nil.